src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisor.scala

```diff
@@ -86,19 +86,25 @@ class ClusterMonitorSupervisor(monitorConfig: MonitorConfig, dataprocConfig: Dat
     case RecreateCluster(cluster) =>
       if (monitorConfig.recreateCluster) {
         logger.info(s"Recreating cluster ${cluster.projectNameString}...")
-        val clusterRequest = ClusterRequest(
-          Option(cluster.labels),
-          cluster.jupyterExtensionUri,
-          cluster.jupyterUserScriptUri,
-          Some(cluster.machineConfig),
-          None,
-          cluster.userJupyterExtensionConfig,
-          if (cluster.autopauseThreshold == 0) Some(false) else Some(true),
-          Some(cluster.autopauseThreshold),
-          cluster.defaultClientId,
-          cluster.clusterImages.find(_.tool == Jupyter).map(_.dockerImage))
-        leonardoService.internalCreateCluster(cluster.auditInfo.creator, cluster.serviceAccountInfo, cluster.googleProject, cluster.clusterName, clusterRequest).failed.foreach { e =>
-          logger.error(s"Error occurred recreating cluster ${cluster.projectNameString}", e)
+        dbRef.inTransaction { dataAccess =>
+          dataAccess.clusterQuery.getClusterById(cluster.id)
+        }.flatMap {
+          case Some(cluster) => val clusterRequest = ClusterRequest(
+            Option(cluster.labels),
+            cluster.jupyterExtensionUri,
+            cluster.jupyterUserScriptUri,
+            Some(cluster.machineConfig),
+            None,
+            cluster.userJupyterExtensionConfig,
+            if (cluster.autopauseThreshold == 0) Some(false) else Some(true),
+            Some(cluster.autopauseThreshold),
+            cluster.defaultClientId,
+            cluster.clusterImages.find(_.tool == Jupyter).map(_.dockerImage))
+            leonardoService.internalCreateCluster(cluster.auditInfo.creator, cluster.serviceAccountInfo, cluster.googleProject, cluster.clusterName, clusterRequest).failed.foreach { e =>
+              logger.error(s"Error occurred recreating cluster ${cluster.projectNameString}", e)
+            }
+            Future.successful(())
```

Comment:
Indentation is a bit off. Also I don't think you need the final `Future.successful(())` -- can just return the result of `leonardoServer.internalCreateCluster`.